src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -62,6 +64,15 @@ public sealed class DiagnosticDescriptor : IEquatable<DiagnosticDescriptor?>
         /// </summary>
         public IEnumerable<string> CustomTags { get; }
 
+        internal ImmutableArray<string> ImmutableCustomTags
+        {
+            get
+            {
+                Debug.Assert(CustomTags is ImmutableArray<string>);
+                return Unsafe.Unbox<ImmutableArray<string>>(CustomTags);
```

Comment:
=-o.  love it

Reply:
Why is this preferrable to casting?

Reply:
The cast, that will fail in release mode, seems significantly better to me than blindly assuming anything.

Reply:
More I'm just not sure what advantage `Unbox` gives here. Typically I think of `Unbox` as useful when I want a `ref T` for the unbox. Essentially the ability to mutate the object. Here we immediately return the result by value hence I'm just not clear what `Unsafe.Unbox` is giving us here. 

Not saying it's wrong, just saying I'm missing the win it's providing over casting.

Reply:
In this case a cast would be fine. I'll fix that.

Reply:
➡️ Switched to cast operator and fixed the build break.